File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/domain/entity/AuthorEntity.java
package ru.spring5.elena.librarymanager.domain.entity;

import lombok.*;
import org.hibernate.annotations.Formula;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;

import static javax.persistence.GenerationType.IDENTITY;

@Entity
@Table(name = "AUTHORS")
@Getter
@NoArgsConstructor
//@RequiredArgsConstructor
@Setter
public class AuthorEntity {
    @Id
    @GeneratedValue(strategy = IDENTITY) //уникальный ID, генерируется сам в БД
    @Column(name = "AUTHOR_ID")
    @NotNull
    @Setter(AccessLevel.NONE)
    private Long authorId;

    @Version
    @Column(name = "VERSION")
    private int version; //версия, инкрементится при редактировании записи

    @NotEmpty(message = "{validation.firstname.notempty.message}")
    @Size(min = 3, max = 60, message = "{validation.firstname.size.message}")
    @Column(name = "FIRST_NAME")
    private String firstName; //имя

    @NotEmpty(message = "{validation.surname.notempty.message}")
    @Size(min = 1, max = 60, message = "{validation.surname.size.message}")
    @Column(name = "SURNAME")
    private String surname; //фамилия

    @Size(min = 0, max = 60, message = "{validation.patronymic.Size.message}")
    @Column(name = "PATRONYMIC")
    private String patronymic; //отчество

    @NotNull(message = "{validation.birthday.NotEmpty.message}")
    @Temporal(TemporalType.DATE)
    @Column(name = "BIRTHDATE")
    private Date birthdate; //

    @Temporal(TemporalType.DATE)
    @Column(name = "DEATHDATE")
    private Date deathdate; //

    @Size(min = 0, max = 1000, message = "{validation.info.Size.message}")
    @Column(name = "INFO")
    private String info; //доп. информация

    @ManyToMany
    @JoinTable(name = "AUTHOR_EDITION", joinColumns = @JoinColumn(name = "AUTHOR_ID"), inverseJoinColumns = @JoinColumn(name = "EDITION_ID"))
    @Setter(AccessLevel.NONE)
    private Set<EditionEntity> editions = new HashSet<EditionEntity>(); //Все работы данного автора

    @Formula("(SURNAME || ' ' || FIRST_NAME || ' ' || PATRONYMIC)")
    @Transient
    private String fio;
}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/domain/entity/AuthorEntity.kt
package ru.elenacherev.librarymanager.domain.entity

import org.hibernate.annotations.ColumnDefault
import org.hibernate.annotations.Formula
import java.time.Instant
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.JoinTable
import javax.persistence.ManyToMany
import javax.persistence.Table
import javax.persistence.Temporal
import javax.persistence.TemporalType
import javax.persistence.Transient
import javax.persistence.Version
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

@Entity
@Table(name = "AUTHORS")
data class AuthorEntity(

    @Column(name = "AUTHOR_ID")
    @GeneratedValue
    @Id
    @get:NotNull
    var authorId: UUID? = null,

    //версия, инкрементится при редактировании записи
    @Column(name = "VERSION", nullable = false)
    @Version
    var version: Int = 0,

    //имя
    @Column(name = "FIRST_NAME", nullable = false)
    @NotEmpty
    @Size(min = 3, max = 60)
    var firstName: String,

    //фамилия
    @Column(name = "SURNAME", nullable = false)
    @NotEmpty
    @Size(min = 1, max = 60)
    var surname: String,

    //отчество
    @Column(name = "PATRONYMIC")
    @Size(min = 0, max = 60)
    var patronymic: String? = null,

    @Column(name = "BIRTHDATE")
    @NotNull
    var birthdate: Instant? = null,

    @Column(name = "DEATHDATE")
    var deathdate: Instant? = null,

    //доп. информация
    @Column(name = "INFO")
    @Size(min = 0, max = 1000)
    var info: String? = null,

    //Все работы данного автора
    @ManyToMany(targetEntity = EditionEntity::class)
    @JoinTable(
        name = "AUTHORS_EDITIONS",
        joinColumns = [JoinColumn(name = "AUTHOR_ID")],
        inverseJoinColumns = [JoinColumn(name = "EDITION_ID")]
    )
    val editions: MutableSet<EditionEntity> = mutableSetOf()
) {
    @Formula("(SURNAME || ' ' || FIRST_NAME || ' ' || PATRONYMIC)")
    @Transient
    lateinit var fio: String
}